code/modules/spells/spell.dm

```diff
@@ -71,9 +71,10 @@ var/list/spells = typesof(/obj/effect/proc_holder/spell) //needed for the badmin
 		if(!(src in user.mob_spell_list))
 			return 0
 
-	if(user.z == ZLEVEL_CENTCOM && !centcom_cancast) //Certain spells are not allowed on the centcom zlevel
+	var/turf/T = get_turf(user)
+	if(T.z == ZLEVEL_CENTCOM && !centcom_cancast) //Certain spells are not allowed on the centcom zlevel
```

Comment:
Because hulk wall-punches let you get the gear PLACED TANTALIZINGLY OUT OF REACH. its like if you left a Reticence parked next to the syndicate glass walkway titled NT-blaster 9000
and described it as the coolest thing